ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/loader.html

```diff
@@ -12,11 +12,247 @@
 -->
 <!DOCTYPE html>
 <html lang="en">
+
 <head>
     <meta charset="UTF-8">
     <title>Title</title>
+
+    <script type="text/javascript">
+
+        class KeycloakLoader {
+            /**
+             * Load keycloak settings
+             */
+            loadKeycloakSettings() {
+                const msg = "Cannot load keycloak settings. This is normal for single-user mode.";
+
+                return new Promise((resolve, reject) => {
+                    try {
+                        const request = new XMLHttpRequest();
+
+                        request.onerror = request.onabort = function () {
+                            reject(msg);
+                        };
+
+                        request.onload = () => {
+                            if (request.status == 200) {
+                                resolve(this.injectKeycloakScript(JSON.parse(request.responseText)));
+                            } else {
+                                reject(null);
+                            }
+                        };
+
+                        const url = "/api/keycloak/settings";
+                        request.open("GET", url, true);
+                        request.send();
+                    } catch (e) {
+                        reject(msg + e.message);
+                    }
+                });
+            }
+
+            /**
+             * Injects keycloak javascript
+             */
+            injectKeycloakScript(keycloakSettings) {
+                return new Promise((resolve, reject) => {
+                    const script = document.createElement('script');
+                    script.type = 'text/javascript';
+                    script.language = 'javascript';
+                    script.async = true;
+                    script.src = keycloakSettings['che.keycloak.js_adapter_url'];
+
+                    script.onload = () => {
+                        resolve(this.initKeycloak(keycloakSettings));
+                    };
+
+                    script.onerror = script.onabort = () => {
+                        reject('Cannot load ' + script.src);
+                    };
+
+                    document.head.appendChild(script);
+                });
+            }
+
+            /**
+             * Initialize keycloak and load the IDE
```

Comment:
Does it really load IDE or it is copy/paste error?

Reply:
Yes, that's copy/paste, there is just keycloak initialization.